components/brave_wallet/common/zcash_utils.h

```diff
@@ -89,21 +104,218 @@ struct OrchardOutput {
 };
 
 // Structure describes note nullifier that marks some note as spent
-struct OrchardNullifier {
+struct OrchardNoteSpend {
   // Block id where spent nullifier was met
   uint32_t block_id = 0;
   std::array<uint8_t, kOrchardNullifierSize> nullifier;
 
-  bool operator==(const OrchardNullifier& other) const = default;
+  bool operator==(const OrchardNoteSpend& other) const = default;
 };
 
-// Structure describes found spendable note
+// Describes spendable note.
+// Spendable note contains related position
+// in the Orchard commitment tree, amount and data required
+// for costructing zk-proof for spending.
 struct OrchardNote {
+  OrchardAddrRawPart addr;
   uint32_t block_id = 0;
-  std::array<uint8_t, kOrchardNullifierSize> nullifier;
+  OrchardNullifier nullifier;
   uint32_t amount = 0;
+  uint32_t orchard_commitment_tree_position = 0;
+  OrchardRho rho;
+  OrchardRseed seed;
 
   bool operator==(const OrchardNote& other) const = default;
+  base::Value::Dict ToValue() const;
+  static std::optional<OrchardNote> FromValue(const base::Value::Dict& value);
+};
+
+// Note witness is a Merkle path in the Orchard commitment tree from the
+// note to the tree root according some selected anchor(selected right border in
+// the commitment tree).
+struct OrchardNoteWitness {
```

Comment:
This entire file is a big problem, we should not just be dumping a bunch of structs in here like this. You don't necessarily need a file for each, but they should be organized in some sensible way because this file is a mess

Reply:
it seems like these probably belong with https://github.com/brave/brave-core/pull/26477/files#r1840956570

Reply:
Moved